lib/application/view-models/details/collection_details_vm.dart

```diff
@@ -0,0 +1,104 @@
+import 'dart:async';
+
+import 'package:equatable/equatable.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:memo/application/view-models/app_vm.dart';
+import 'package:memo/application/view-models/item_metadata.dart';
+import 'package:memo/domain/enums/resource_type.dart';
+import 'package:memo/domain/models/resource.dart';
+import 'package:memo/domain/services/collection_services.dart';
+import 'package:memo/domain/services/resource_services.dart';
+import 'package:memo/domain/transients/collection_status.dart';
+
+final collectionDetailsVM = StateNotifierProvider.family<CollectionDetailsVM, String>(
+  (ref, collectionId) => CollectionDetailsVMImpl(
+    collectionId: collectionId,
+    collectionServices: ref.read(collectionServices),
+    resourceServices: ref.read(resourceServices),
+  ),
+);
+
+class ResourceInfo extends Equatable {
+  const ResourceInfo(this.type, {required this.description, required this.url});
+
+  final ResourceType type;
+  final String description;
+  final String url;
+
+  @override
+  List<Object?> get props => [type, description, url];
+}
+
+abstract class CollectionDetailsVM extends StateNotifier<CollectionDetailsState> {
+  CollectionDetailsVM(CollectionDetailsState state) : super(state);
+}
+
+class CollectionDetailsVMImpl extends CollectionDetailsVM {
+  CollectionDetailsVMImpl(
+      {required this.collectionId, required this.collectionServices, required this.resourceServices})
+      : super(LoadingCollectionDetailsState()) {
+    _loadCollection();
+  }
+
+  final String collectionId;
+  final CollectionServices collectionServices;
+  final ResourceServices resourceServices;
+
+  List<Resource>? _associatedResources;
+  late final StreamSubscription<CollectionStatus> _listener;
+
+  Future<void> _loadCollection() async {
+    final stream = await collectionServices.listenToCollectionStatus(collectionId: collectionId);
+
+    _listener = stream.listen((collectionStatus) async {
```

Comment:
I couldn't understand why the details vm listen to the collection status. What's the scenario where the collection status may change while this page is loaded in memory?

Reply:
the user might return to this page, for any possible reason, after executing the same collection

Reply:
Ahh, got it. The execution is presented in a modal